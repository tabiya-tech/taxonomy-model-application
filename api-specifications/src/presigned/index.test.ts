describe("Test the presigned module", () => {
  test("The presigned module can be required via the index", async () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => await import("./")).not.toThrow(); // We check that it doesn't throw an error instead of simply letting it fail on import because we want an easier error message
    const presignedModule = await import("./");

    // THEN check if Schemas is defined in it
    expect(presignedModule.default.Schemas.GET.Response.Payload).toBeDefined();
    // AND check if constants are defined in it
    const Constants = presignedModule.default.Constants;
    expect(Constants.EXPIRES).toBeDefined();
    expect(Constants.MAX_FILE_SIZE).toBeDefined();
  });

  test("The presigned module matches the snapshot", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => {
      // THEN Check if the module can be required without error
      const presignedModule = await import("./");
      expect(presignedModule.default).toMatchSnapshot();
    }).not.toThrowError();
  });
});

export {};
