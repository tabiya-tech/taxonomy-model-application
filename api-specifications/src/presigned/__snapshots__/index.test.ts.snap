// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the presigned module The presigned module matches the snapshot 1`] = `
{
  "Constants": {
    "EXPIRES": 3600,
    "MAX_FILE_SIZE": 20971520,
    "MAX_FOLDER_NAME_LENGTH": 256,
    "MAX_URI_LENGTH": 4096,
  },
  "Schemas": {
    "GET": {
      "Response": {
        "Payload": {
          "$id": "/components/schemas/PresignedResponseSchemaGET",
          "additionalProperties": false,
          "properties": {
            "fields": {
              "description": "The fields should be added to the form-data when uploading the files",
              "items": {
                "properties": {
                  "name": {
                    "pattern": "\\S",
                    "type": "string",
                  },
                  "value": {
                    "pattern": "\\S",
                    "type": "string",
                  },
                },
                "required": [
                  "name",
                  "value",
                ],
                "type": "object",
              },
              "minItems": 1,
              "type": "array",
            },
            "folder": {
              "description": "The folder name to upload the files to",
              "maxLength": 256,
              "pattern": "\\S",
              "type": "string",
            },
            "url": {
              "description": "The url to upload the files to",
              "format": "uri",
              "maxLength": 4096,
              "pattern": "^https://.*",
              "type": "string",
            },
          },
          "required": [
            "url",
            "fields",
            "folder",
          ],
          "type": "object",
        },
      },
    },
  },
  "Types": {},
}
`;
